src/components/RecipeForm/RecipeForm.jsx

```diff
@@ -0,0 +1,109 @@
+import { useState, useEffect } from 'react';
+import styles from './RecipeForm.module.css';
+import { RecipeFormTextarea } from './RecipeFormTextarea';
+
+export const RecipeForm = ({ onAddRecipe, editingRecipe }) => {
+  const [ingredients, setIngredients] = useState('');
+  const [allergens, setAllergens] = useState('');
+  const [cookingSteps, setCookingSteps] = useState('');
+  const [photo, setPhoto] = useState('');
+  const [photoFile, setPhotoFile] = useState('');
+  const [isEditing, setIsEditing] = useState(false);
+
+  useEffect(() => {
```

Comment:
Avoid using `useEffect` as much as possible. Before using it, always think twice, asking yourself if you really 100% need it.

If you want the parent component to control some parts of the state of `RecipeForm`, put the state into the parent.

Soon, we will learn various ways of maintaining a global state that can be accessed by all components without having to provide it through props.